### PR TITLE
Functional test coverage for Vue2→Vue3 migration + real bug fixes (#19772)

### LINKED ISSUE #19772: Track missing functional test coverage for the ES-module/TypeScript editor refactor
## Context

PR #19522 (and the branches it stacks on, #19489/#19677/#19682) converts a large number of dynamically-injected field/part editor scripts from inline `onDocumentReady`-wrapped jQuery to real ES modules / TypeScript. That migration class already produced one real regression caught only by manual testing (the AJAX-widget double-init bug fixed by #19442, and a near-miss in this same PR — see the Features admin page bug fixed in commit `f5533d29bd`, where three attribute/value mismatches silently broke bulk-action filtering with zero automated test catching it before manual QA).

Given the size of the refactor (241+ TypeScript files across ~58 modules), a full agent-assisted triage was run to categorize every touched module by regression risk and cross-reference it against existing Playwright functional test coverage (`test/OrchardCore.Tests.Functional/Tests/Cms/*.cs`).

**Categories used:**
- **A — Rewritten logic**: real behavior risk (new selectors/event wiring, jQuery removed, AJAX-reachability changed, conditional show/hide logic rewritten).
- **B — Mechanical transpile**: same behavior, just moved to TS/bundled differently. Low risk.
- **C — No interactive JS**: settings/config forms with only simple field bindings. Skippable.

This issue tracks the **Category A modules with no or only partial functional test coverage** — the modules where a regression could currently ship completely undetected by CI.

## Already covered (for reference, no action needed)
- `OrchardCore.Flows` — `FlowsWidgetDragTests.cs`, `FlowPartWidgetLifecycleTests.cs`
- `OrchardCore.Templates` / `OrchardCore.Liquid` — `MonacoLiquidIntelliSenseTests.cs`
- `OrchardCore.Html` (HtmlBodyPart Wysiwyg/Trumbowyg) — `EsModuleEditorTests.cs`
- `OrchardCore.Shortcodes` — `MonacoLiquidIntelliSenseTests.cs` + `BulkSelectListTests.cs`
- 12 admin list views (Contents, Placements, Shortcodes, Sitemaps×2, Templates, FeatureProfiles, UrlRewriting, DeploymentPlan, MediaProfiles, AdminMenu, Queries) — `BulkSelectListTests.cs`
- `OrchardCore.Tenants` — `TenantsBulkSelectTests.cs`
- `OrchardCore.AdminDashboard` (drag-reorder only) — `AdminDashboardDragTests.cs`
- `OrchardCore.AdminMenu` (node drag/reorder only) — `AdminMenuTreeTests.cs`
- `OrchardCore.Menu`, `OrchardCore.Taxonomies` (hierarchy only) — `MenuHierarchyTests.cs`, `TaxonomyHierarchyTests.cs`
- `OrchardCore.Layers` — `LayersDragTests.cs`
- `OrchardCore.Markdown` — `MarkdownEditorTests.cs`
- `OrchardCore.Media` — `MediaAppTests.cs`, `MediaSecurityTests.cs`, `MediaTus*Tests.cs`
- `OrchardCore.Widgets` — `WidgetsListDragTests.cs`

## Gaps to close

Each item below is independently actionable — pick any one, no ordering dependency unless noted.

- [ ] **OrchardCore.Forms** — `SelectPart`/`InputPart`/`FormElementLabelPart` editors are addable via the Flows/Bag AJAX "Add Widget" flow, exactly the double-instance-init bug class #19442 fixed. AJAX-add two `SelectPart` (or `InputPart`) widget instances, assert both initialize independently (e.g. both show populated option rows / both react to the type dropdown).
- [ ] **OrchardCore.Notifications** — `user-notification-preferences-part.ts` reimplements custom HTML5 drag-and-drop reorder outside the shared bloom components. Drag-reorder the "Methods" list on the user notification preferences page, assert new order persists after save.
- [ ] **OrchardCore.OpenId** — `openid-server-settings.ts` (214 lines) / `openid-client-settings.ts` (70 lines) are large hand-rewritten conditional-visibility/collapse state machines (flow checkboxes toggling other fields, certificate-store cascades). Toggle each allowed-flow checkbox on `/Admin/OpenId/Application/Create`, assert dependent fields show/hide correctly.
- [ ] **OrchardCore.Seo** — `SeoMetaPart` editor (custom meta tags modal + Google Schema JSON-LD CodeMirror field) is AJAX-injectable via Bag/Flow widgets. AJAX-inject the part editor, edit both the custom-meta-tags modal and JSON-LD field, assert both persist — specifically verifying `observeAndInit` re-fires for the AJAX-injected instance.
- [ ] **OrchardCore.Cors** — `cors-admin.js` Vue app had jQuery fully replaced with native DOM APIs and `corsApp` promoted to `window.corsApp`. Add/edit a CORS policy (origins/methods) via the admin settings page, assert it persists.
- [ ] **OrchardCore.DataLocalization** — `translation-editor.ts` (285 lines) is a net-new interactive Vue translation editor, not a straight port. Select a culture, edit a translation string, save, assert it persists on reload.
- [ ] **OrchardCore.Deployment** — `steporder.js` step reordering rewritten from jQuery `$.ajax` to fetch-based DOM APIs. Reorder deployment plan steps, assert the new order persists.
- [ ] **OrchardCore.AzureAI** — `azure-ai-search-default-settings.ts` converts an inline auth-type-dropdown-toggle script to a module. Switch the auth-type select on the Azure AI Search default connection settings page, assert the correct dependent section shows/hides.
- [ ] **OrchardCore.Features** — `features-index.ts` (218 lines) is a fully rewritten interactive admin script (search/filter, per-group select-all, badge show-more, keyboard Escape handling). This exact class of bug already shipped once and was only caught manually (see `f5533d29bd` on PR #19522) — no automated test exists. Search for a feature, use select-all within a group, assert correct checkboxes get checked.
- [ ] **OrchardCore.Recipes** — `recipes-index.ts` reimplements client-side search/filter by hand instead of using shared list-management components. Type a partial recipe name into the search box, assert the list filters correctly.
- [ ] **OrchardCore.ContentTypes** — `content-types-sitemap-source.ts` (90 lines, cascading show/hide/enable-disable logic), `content-types-admin-node.ts` (checkbox-driven card toggling + icon picker). Drag-reorder a field/part on `/Admin/ContentTypes/Edit/{type}` and assert order persists; separately, toggle the sitemap source's "Disabled"/"Manage contained item routes" checkboxes and assert dependent fields respond.
- [ ] **OrchardCore.Users** — `user-fields.ts` (80 net-new lines), `request-verification-code.ts` (31 net-new lines) — substantial new interactive logic with no dedicated test.
- [ ] **OrchardCore.Alias** — AJAX-reachable action toggling "Disabled"/"Manage contained item routes" checkboxes with dependent field show/hide; no test found.

### Partial coverage — extend existing tests
- [ ] **OrchardCore.AdminDashboard** — `AdminDashboardDragTests.cs` covers reorder only. Add a resize-handle test (`.dashboard-resize-handle-se`, new pointer-based snap-to-grid math) asserting `--dashboard-width`/`--dashboard-height` persist after reload.
- [ ] **OrchardCore.AdminMenu** — `AdminMenuTreeTests.cs` covers drag/reorder only. Add a test opening the Link/Placeholder admin-node edit modal (AJAX-injected) and exercising the Vue icon-picker and permission-picker widgets (rewired with `self`→`this` binding changes — real regression risk).
- [ ] **OrchardCore.Sitemaps** — `BulkSelectListTests.cs` covers the list view only. `sitemap-cache-list.ts` (40 net-new lines) adds new cache-clear/refresh interactions with no coverage.
- [ ] **OrchardCore.UrlRewriting** — `BulkSelectListTests.cs` covers the list view only. `url-rewriting-admin-index.ts` (21 net-new lines) + `sortable-rules.js` reorder/toggle interactions are untested.

## Shared infrastructure — highest blast radius, zero direct/isolated coverage

These aren't module-specific but underpin many of the modules above; a regression here would break several modules silently at once. Lower priority than the module-specific items above (each already gets *some* indirect exercise from whichever consumer test happens to run), but worth a dedicated test given the blast radius:

- [ ] `helpers/observeAndInit.ts` — the generic AJAX-reinit helper nearly every editor/field script wires through (Flows, Bag, Widgets-list, Content Fields, Menu, etc.). No dedicated test.
- [ ] `helpers/monaco.ts` + `helpers/monacoTheme.ts` — shared Monaco bootstrap/theming used by Templates, Queries, Workflows scripting, ContentFields JSON/text-Monaco, Elasticsearch/Lucene query editors, OpenId application editor. Only indirectly covered via `MonacoLiquidIntelliSenseTests.cs` for one consumer.
- [ ] `components/sortable-menu.ts` — largest single new component (297 lines), shared by Menu and Taxonomies-style hierarchical/sortable admin trees. `MenuHierarchyTests.cs`/`TaxonomyHierarchyTests.cs` test the consumer UI end-to-end but not this component in isolation.
- [ ] `components/openid-application-editor.ts` — very large (319 new lines), single consumer (OpenId) but highest rewrite-risk-per-line in the whole refactor.
- [ ] `components/bulk-select-list.ts` + `components/workflow-bulk-select-list.ts` — used by 8+ admin list views; only indirectly exercised via `BulkSelectListTests.cs`/`TenantsBulkSelectTests.cs`, the component's own selection/keyboard/edge-case logic has no isolated test.

## Notes for whoever picks an item up
- Follow the patterns already established in `test/OrchardCore.Tests.Functional/Tests/Cms/`: `EsModuleEditorTests.cs` for the AJAX-injection double-init pattern, `BulkSelectListTests.cs` for a recipe-seeded multi-view theory test, `MonacoLiquidIntelliSenseTests.cs` for driving Monaco's real keyboard/suggestion-widget behavior, `FlowPartWidgetLifecycleTests.cs` for widget add/collapse/delete via TheAdmin's `confirmDialog()`.
- Most of these views need a recipe step to seed data — check `test/OrchardCore.Tests.Functional/Fixtures/*.recipe.json` for an existing example before writing a new one from scratch.
- Fast/cheap tests (~8s) can go straight into the `Cms` namespace and ride along in `pr_ci.yml`'s existing `--filter-class "*Cms*"` step on every PR push. Larger multi-view suites should get their own low-frequency workflow like `functional_bulk_select_list.yml` (manual dispatch + on-review + push-to-main only) to avoid bloating per-commit CI time.

### PR DESCRIPTION
**Pull request by Skrypt**
## Summary

Adds functional test coverage for the Vue 2 → Vue 3 / ES-module migration (#19774), per the plan in issue #19772. Fixes 6 real production bugs discovered along the way while writing tests, in addition to landing 17 new/extended functional test cases.

Base branch: `skrypt/yarn-check` (open PR #19489, the umbrella jQuery-removal/ES-module/Vue3 effort this stacks on top of).

## Real bugs found and fixed while writing tests

| Bug | Commit | Impact |
|---|---|---|
| `Features` admin group select-all/summary selector mismatch | `9366e6e866` | Group checkbox UI broken after ES-module conversion |
| `vuedraggable` UMD global unwrapping broken in 3 consumers | `5ff3f8f645` | Drag-reorder silently non-functional in 3 places |
| OpenId client settings script `depends-on` attribute wrong | `189c2548c0` | Script load-order bug, settings editor broken |
| `ContentTypesAdminNodeDataLocalizationProvider` not guarded behind `AdminMenu` feature | `1931297d4a` | Crash when DataLocalization enabled without AdminMenu |
| **16-file cross-module hardcoded-id bug** (Alias, ContentPreview, ContentFields, Title, Autoroute, Contents, Sms, Email, Tenants, RateLimits, Email.Smtp, Elasticsearch, ContentTypes, Notifications) | `0a6ffa9d6a` | CodeMirror pattern editors and multiple dependent-field toggles silently broken across 13 modules — the largest fix in this PR, root-caused to a historical refactor predating this migration |
| Sitemaps cache page missing the `list-management.js` script include entirely | `a5f30d6c91` | Search box on `/Admin/SitemapsCache/List` did nothing for every user |

Each bug was found via genuine TDD red/green verification (temporarily reverting the fix, confirming the new test fails for the right reason, then restoring the fix and confirming it passes) — not assumed from reading code.

## New/extended test coverage

- Features admin group checkboxes
- Forms `SelectPart` (3 consumers)
- OpenId client settings script
- Seo `SeoMetaPart` custom meta tags editor
- Cors policy add/edit persistence
- DataLocalization translation editor
- Deployment step reorder persistence
- AzureAI search auth-type toggle
- Recipes admin search/filter
- ContentTypes editor reorder + sitemap-source toggles
- Users password generator field editor
- Alias pattern editor (CodeMirror) — surfaced the 16-file id bug above
- Notifications preferences drag-reorder
- AdminDashboard widget resize-handle
- AdminMenu permission-picker
- Sitemaps cache search + Purge/Purge All — surfaced the missing-script bug above
- UrlRewriting rule drag-reorder

## Explicitly out of scope / skipped

- **Task 18** (`observeAndInit.ts` isolated unit test): no JS unit-test framework (Vitest or otherwise) exists anywhere in this repo; introducing one for a single 94-line helper was judged disproportionate. The helper gets solid indirect coverage via every consumer already exercised by the Playwright tests above.
- **Tasks 19–21** (Monaco 2nd-consumer, sortable-menu.ts isolated test, bulk-select-list.ts edge cases): lower-priority stretch goals per the original issue's own prioritization, not pursued in this PR.

## Verification

- `yarn lint && yarn check`: clean across the whole monorepo, run after every task.
- Full `*Cms*` functional suite: consistently 108/109–112 depending on run, with only pre-existing/environmental flakes (SaasTests locale timing, TaxonomyHierarchyTests drag timing, Monaco IntelliSense timing under load, and inotify-instance-limit exhaustion from running many parallel fixtures across a long session) — never caused by this PR's changes, confirmed by rerunning the specific new/changed test classes in isolation.
- Every new test was verified via a genuine TDD red/green cycle against the real fix, not just "does it pass once."

Fixes #19772

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Localization/Models/LocalizationSettings.cs [hunk 1]
```diff
@@ -7,14 +7,22 @@ namespace OrchardCore.Localization.Models;
 /// </summary>
 public class LocalizationSettings
 {
-    private static readonly string[] s_defaultSupportedCultures = [CultureInfo.InstalledUICulture.Name];
+    // CultureInfo.InstalledUICulture.Name is empty when the OS/runtime has no configured UI
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Localization/Models/LocalizationSettings.cs:10]: @hishamco please take a look at this. I've had a lot of issues with Playwright tests because of this. Now, I believe this used to be fine and something removed the fallback lines over time.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Localization/Models/LocalizationSettings.cs:10]: The issue is that it is not given that an OS has a valid InstalledUICulture that is reflected within the .NET Core framework. 
This is why we need an hard coded fallback culture.

**Comment by Skrypt** [comment]
@sebastienros You've got pretty much everyting related with javascript refactoring on both of these branches. There's 2 pull requests for them. Only thing remaining is to do some manual tests.

File: src/OrchardCore.Modules/OrchardCore.Localization/Models/LocalizationSettings.cs (hunk 1 above)
Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Localization/Models/LocalizationSettings.cs:10]: LGTM
